### PR TITLE
Sponsorship: the service was added

### PR DESCRIPTION
**Pull request by asamal**

### CODE REVIEW

**Comment by asamal** [comment]
Just added all previous commits as a pull request

File: sponsorship/build.gradle [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	}
+}
+
+repositories {
+	mavenCentral()
+}
+
+dependencies {
+	implementation 'org.springframework.boot:spring-boot-starter-actuator'
+	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (xpinjection) [inline sponsorship/build.gradle]: We don't need JPA at that stage.

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: There is no reason for onConstructor param

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+@RequestMapping(value = "/api/v1/sponsorship/plan",
+    consumes = { MediaType.APPLICATION_JSON_UTF8_VALUE }, produces = { MediaType.APPLICATION_JSON_UTF8_VALUE })
+@Api(value = "Register a new sponsorship plan")
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java:34]: This API could be used not only for plan registration

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+public class PlanController
+{
+  private final PlanService planService;
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java:37]: We don't need implementation at that stage

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+
+  @ApiOperation(value = "Register a new sponsorship plan", response = PlanRegistered.class)
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: There is no way to specify response, Swagger could get this info from Spring MVC annotations

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  @PostMapping("/register")
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: It have to be POST on .../plans URL

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "OK", response = PlanRegistered.class),
+      @ApiResponse(code = 401, message = "Failed", response = String.class)
+  })
+  ResponseEntity<PlanRegistered> register(@RequestBody PlanRegistrationRequest plan)
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: There is no reason to wrap in response entity.

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  {
+    return ResponseEntity.ok(new PlanRegistered(planService.registerPlan(plan).getId()));
+  }
+
+
+  @ApiOperation(value = "Cancel the sponsorship plan", response = PlanCancelled.class)
+  @PostMapping("/cancel")
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: It must be PUT on .../plans/{id}/cancellation or .../plans/{id}/status

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+public class PlanController
+{
+  private final PlanService planService;
+
+
+  @ApiOperation(value = "Register a new sponsorship plan", response = PlanRegistered.class)
+  @PostMapping("/register")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "OK", response = PlanRegistered.class),
+      @ApiResponse(code = 401, message = "Failed", response = String.class)
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: It is important to explain what does this code mean. 401 is strange code in this context.

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+
+  @ApiOperation(value = "Cancel the sponsorship plan", response = PlanCancelled.class)
+  @PostMapping("/cancel")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "OK", response = PlanCancelled.class),
+      @ApiResponse(code = 401, message = "Failed", response = String.class)
+  })
+  ResponseEntity<PlanCancelled> cancel(@RequestBody PlanCancellationRequest plan)
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: Usually on update latest state of resource is returned.

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@RequestMapping(value = "/api/v1/sponsorship/sponsor",
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java]: Common prefix could be moved to common base class together with produces and consumes

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    consumes = { MediaType.APPLICATION_JSON_UTF8_VALUE }, produces = { MediaType.APPLICATION_JSON_UTF8_VALUE })
+@Api(value = "Register a new sponsor")
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java:29]: Probably, not only this operation have to be supported

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+public class SponsorController
+{
+  private final SponsorService sponsorService;
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java:32]: No implementation is needed at that stage

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+
+  @ApiOperation(value = "Register a new sponsor", response = SponsorRegistered.class)
+  @PostMapping("/register")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "OK", response = SponsorRegistered.class),
+      @ApiResponse(code = 401, message = "Failed", response = String.class)
+  })
+  ResponseEntity<SponsorRegistered> register(@RequestBody SponsorRegistrationRequest sponsor)
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java]: The same set of issues as for other methods in controllers.

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanCancellationRequest.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@Data
+@Builder
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanCancellationRequest.java]: Builder is useless

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@RequestMapping(value = "/api/v1/sponsorship/plan",
+    consumes = { MediaType.APPLICATION_JSON_UTF8_VALUE }, produces = { MediaType.APPLICATION_JSON_UTF8_VALUE })
+@Api(value = "Register a new sponsorship plan")
+public class PlanController
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java:35]: No data validation applied

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@RequestMapping(value = "/api/v1/sponsorship/sponsor",
+    consumes = { MediaType.APPLICATION_JSON_UTF8_VALUE }, produces = { MediaType.APPLICATION_JSON_UTF8_VALUE })
+@Api(value = "Register a new sponsor")
+public class SponsorController
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java:30]: No data validation applied.

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@Data
+@Builder
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java]: Builder is useless

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+public class PlanRegistrationRequest
+{
+  private PlanCategory category;
+  private String sponsorId;
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java]: Why does plan have concrete sponsor?

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/SponsorRegistrationRequest.java [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@Data
+@Builder
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/SponsorRegistrationRequest.java]: Builder is useless

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/configuration/SwaggerConfiguration.java [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+ * @since v1.0
+ */
+@Configuration
+@EnableSwagger2
+public class SwaggerConfiguration
+{
+  @Bean
+  public Docket productApi()
+  {
+    return new Docket(DocumentationType.SWAGGER_2)
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/configuration/SwaggerConfiguration.java:39]: Add global API info and apply filter at URL level

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/domain/Plan.java [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@Data
+@Builder
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/domain/Plan.java:14]: Builder is useless

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/domain/Sponsor.java [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@Data
+@Builder
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/domain/Sponsor.java:14]: Builder is useless

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/PlanService.java [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.Collection;
+import java.util.Map;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@Service
+public class PlanService
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/PlanService.java:11]: Service is not needed at that stage.

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/SponsorService.java [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
+import com.kyriba.conference.sponsorship.domain.Sponsor;
+import org.springframework.stereotype.Service;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@Service
+public class SponsorService
```
Comment (xpinjection) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/SponsorService.java:11]: Service is not needed at that stage.

File: sponsorship/src/test/java/com/kyriba/conference/sponsorship/SponsorshipApplicationTests.java [hunk 26]
```diff
@@ -0,0 +1,16 @@
+package com.kyriba.conference.sponsorship;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class SponsorshipApplicationTests {
```
Comment (xpinjection) [inline sponsorship/src/test/java/com/kyriba/conference/sponsorship/SponsorshipApplicationTests.java]: Useless test

File: sponsorship/src/test/java/com/kyriba/conference/sponsorship/api/PlanControllerTest.java [hunk 27]
```diff
@@ -0,0 +1,61 @@
...
+import static io.restassured.RestAssured.given;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class PlanControllerTest
```
Comment (xpinjection) [inline sponsorship/src/test/java/com/kyriba/conference/sponsorship/api/PlanControllerTest.java]: Where is service documentation with usage samples?

File: sponsorship/src/test/java/com/kyriba/conference/sponsorship/api/SponsorControllerTest.java [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+import static io.restassured.RestAssured.given;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class SponsorControllerTest
```
Comment (xpinjection) [inline sponsorship/src/test/java/com/kyriba/conference/sponsorship/api/SponsorControllerTest.java]: Where is service documentation with usage samples?

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java (hunk 19 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java]: Lets suppose, there are several sponsorship plans available. When some sponsor chooses a plan and pays for it, then the plan becomes reserved for this sponsor. All sponsorship plans with an empty sponsorID are available for reservation. One sponsor can buy several plans. So it is one-to-many relationship and it is logical to have reference to a sponsor from a plan.

File: sponsorship/build.gradle (hunk 1 above)
Comment (asamal) [reply sponsorship/build.gradle]: Removed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 2 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: Removed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 5 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: Fixed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 6 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: Fixed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 7 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: Fixed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 8 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: Done

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 9 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: Fixed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 10 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: Fixed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanCancellationRequest.java (hunk 15 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanCancellationRequest.java]: Removed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java (hunk 18 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java]: Removed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/SponsorRegistrationRequest.java (hunk 20 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/SponsorRegistrationRequest.java]: Removed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/configuration/SwaggerConfiguration.java (hunk 21 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/configuration/SwaggerConfiguration.java:39]: Fixed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/domain/Plan.java (hunk 22 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/domain/Plan.java:14]: Removed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/domain/Sponsor.java (hunk 23 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/domain/Sponsor.java:14]: Removed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 4 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java:37]: I believe, it is similar to https://github.com/gariklolkin/conference/pull/10#discussion_r286013126, so for now, I'd prefer to leave it as is

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/PlanService.java (hunk 24 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/PlanService.java:11]: I believe, it is similar to https://github.com/gariklolkin/conference/pull/10#discussion_r286013126, so for now, I'd prefer to leave it as is

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/SponsorService.java (hunk 25 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/SponsorService.java:11]: I believe, it is similar to https://github.com/gariklolkin/conference/pull/10#discussion_r286013126, so for now, I'd prefer to leave it as is

File: sponsorship/src/test/java/com/kyriba/conference/sponsorship/SponsorshipApplicationTests.java (hunk 26 above)
Comment (asamal) [reply sponsorship/src/test/java/com/kyriba/conference/sponsorship/SponsorshipApplicationTests.java]: Removed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java (hunk 11 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java]: "produces and consumes" - what do you mean?

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java (hunk 13 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java:32]: I believe, it is similar to https://github.com/gariklolkin/conference/pull/10#discussion_r286013126, so for now, I'd prefer to leave it as is

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java (hunk 14 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java]: Done

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 16 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java:35]: Done

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java (hunk 17 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java:30]: Done

File: sponsorship/build.gradle [hunk 29]
```diff
@@ -0,0 +1,33 @@
+plugins {
```
Comment (ilya-a87) [inline sponsorship/build.gradle:1]: asciidoc is missed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 30]
```diff
@@ -0,0 +1,68 @@
...
+{
+  private final PlanService planService;
+
+
+  @SuppressWarnings("unused")
+  @ApiOperation(value = "Register a new sponsorship plan")
+  @PostMapping
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "OK"),
+      @ApiResponse(code = 401, message = "Failed to register a new plan", response = String.class)
```
Comment (ilya-a87) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: `, response = String.class`
I doubt that is should be String

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java [hunk 31]
```diff
@@ -0,0 +1,68 @@
...
+
+
+  @SuppressWarnings("unused")
+  @ApiOperation(value = "Cancel the sponsorship plan")
+  @PutMapping("/{id}/cancellation")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "OK", response = String.class),
+      @ApiResponse(code = 401, message = "Failed to cancel the plan", response = PlanRegistered.class)
+  })
+  PlanRegistered cancel(@ApiParam(value = "Id of the plan to cancel", required = true) @PathVariable String id)
```
Comment (ilya-a87) [inline sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: I think status code in response will be enough, also 'PlanRegistered' on cancellation looks weird

File: sponsorship/build.gradle (hunk 29 above)
Comment (asamal) [reply sponsorship/build.gradle:1]: added

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 30 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: fixed

File: sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java (hunk 31 above)
Comment (asamal) [reply sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java]: fixed

**Pull request merged**